src/java/org/apache/poi/ss/formula/FormulaShifter.java

```diff
@@ -548,4 +556,105 @@ private static Ptg createDeletedRef(Ptg ptg) {
 
         throw new IllegalArgumentException("Unexpected ref ptg class (" + ptg.getClass().getName() + ")");
     }
+    
+    
+    // ******** logic which processes columns in same way as row ********
+    
+
+    /**
+     * @param ptgs - if necessary, will get modified by this method
+     * @param currentExternSheetIx - the extern sheet index of the sheet that contains the formula being adjusted
+     * @return <code>true</code> if a change was made to the formula tokens
+     */
+    public boolean adjustFormula(Ptg[] ptgs, int currentExternSheetIx) {
+        boolean refsWereChanged = false;
+        for(int i=0; i<ptgs.length; i++) {
+            Ptg newPtg;
+            if(_rowModeElseColumn){
+                newPtg = adjustPtg(ptgs[i], currentExternSheetIx);
+                if (newPtg != null) {
+                    refsWereChanged = true;
+                       ptgs[i] = newPtg;
+                }
+            }
+            else {
+                Ptg transposedPtg = transpose(ptgs[i]);
```

Comment:
While column shift = transpose ∘ row shift ∘ transpose, it's an inefficient way to do it.

Reply:
It is, for large enough sheets. But faster solution would must dive deep into adjusting logic, so it would be much harder to implement, and much more error prone. So - at least for this time - it is out of scope of my work.

Reply:
I added FormulaShifter$ShiftMode#ColumnMove to trunk in a recent commit. It was a pretty trivial addition. Unfortunately, the changes required more code duplication than what other languages can offer, but that's Java.